### ISSUE TITLE
How can I pass a document to the envelopes_api's update_document

### ISSUE THREAD

**Issue by tmeinders** [docusign/docusign-python-client#46]
It appears to be missing the actual functionality of uploading a document. I might have a misunderstanding of the purpose of this method, but if it is intended for uploading a document, I can't see how. There is no parameter in the method signature which would include a document payload, and the kwargs are filtered for a set of parameters that don't seem to include a document either. At a glance, I can't see any way that the body of the api call in the called method could be set with a document.

**Comment by LarryKlugerDS** [comment]
You want to use this API method: https://developers.docusign.com/esign-rest-api/reference/Envelopes/EnvelopeDocuments/update ?

I agree with you that the SDK doesn't support this API method correctly. I have filed internal ticket DCM-3778

Meanwhile, you'll need to call the API directly for this functionality.

**Comment by tmeinders** [comment]
Yeah, I understand that I can use the endpoint directly. The method in this client hits that same endpoint, it just doesn't seem to do anything. It appears to be more of a stub in a client that is represented as official (which I'd think would also mean it is complete). Maybe this issue should be marked as a bug or maybe an enhancement rather than a question, even though the original phrasing was in question form because the implementation doesn't make sense. Thanks for opening an internal ticket about this.

**Comment by LarryKlugerDS** [comment]
Hi @tmeinders ,
Yes, the internal ticket, DCM-3778 is a bug report.
We use issue labels question/answered to track if we've responded to the issue or not. I will try to come up with better labels.

The SDK is generated using a fork of the Swagger open source CodeGen app. That app is not correctly generating SDK code for this API call. That's the bug that I've reported for a fix.

**Comment by ronw23** [comment]
I had this exact issue and ending up using the requests library. The ApiClient methods always try to turn a request body into JSON. It is not possible to call PUT endpoints where the body is the raw bytes of a file using the methods in ApiClient.

**Comment by fredthomsen** [comment]
It's disappointing for developers leveraging the SDK that this is still an issue 2 years later.  Made a fix (not sure it's the best way to do it), in my own fork for now as this basically boils down to proper header handling and an additional parameter for the document.

**Comment by aadarshjajodiaSB** [comment]
This issue has been resolved in the current version of the SDK. The update_document method now accepts a required document_file_bytes parameter, which is correctly passed as the request body with Content-Type:
  application/pdf. You no longer need to call the API directly or use a workaround — the method signature is:

  envelopes_api.update_document(account_id, document_id, envelope_id, document_file_bytes)

  Closing this issue. Please reopen if you encounter further problems.
